test/support/test_server.ex

```diff
@@ -89,134 +87,110 @@ defmodule WebSockex.TestServer do
-    {:ok, req, state}
+    {:ok, state}
   end
 
-  def websocket_info(:stall, _, _) do
+  def websocket_info(:stall, _) do
     Process.sleep(:infinity)
   end
 
-  def websocket_info(:send_ping, req, state), do: {:reply, :ping, req, state}
+  def websocket_info(:send_ping, state), do: {:reply, :ping, state}
 
-  def websocket_info(:send_payload_ping, req, state) do
+  def websocket_info(:send_payload_ping, state) do
     payload = "Llama and Lambs"
-    {:reply, {:ping, payload}, req, Map.put(state, :ping_payload, payload)}
+    {:reply, {:ping, payload}, Map.put(state, :ping_payload, payload)}
   end
 
-  def websocket_info(:close, req, state), do: {:reply, :close, req, state}
+  def websocket_info(:close, state), do: {:reply, :close, state}
+
+  def websocket_info({:close, reason}, state) do
+    {:reply, {:close, reason}, state}
+  end
 
-  def websocket_info({:close, code, reason}, req, state) do
-    {:reply, {:close, code, reason}, req, state}
+  def websocket_info({:close, code, reason}, state) do
+    {:reply, {:close, code, reason}, state}
   end
 
-  def websocket_info({:send, frame}, req, state) do
-    {:reply, frame, req, state}
+  def websocket_info({:send, frame}, state) do
+    {:reply, frame, state}
   end
 
-  def websocket_info({:set_code, code}, req, state) do
+  def websocket_info({:set_code, code}, state) do
     Agent.update(state.agent_pid, fn _ -> code end)
-    {:ok, req, state}
+    {:ok, state}
   end
 
-  def websocket_info(:connection_wait, req, state) do
+  def websocket_info(:connection_wait, state) do
     Agent.update(state.agent_pid, fn _ -> :connection_wait end)
-    {:ok, req, state}
+    {:ok, state}
   end
 
-  def websocket_info(:immediate_reply, req, state) do
+  def websocket_info(:immediate_reply, state) do
     Agent.update(state.agent_pid, fn _ -> :immediate_reply end)
-    {:ok, req, state}
+    {:ok, state}
   end
 
-  def websocket_info(:shutdown, req, state) do
-    {:shutdown, req, state}
+  def websocket_info(:shutdown, state) do
+    {:shutdown, state}
   end
 
-  def websocket_info(_, req, state), do: {:ok, req, state}
-
-  @dialyzer {:nowarn_function, immediate_reply: 1}
-  defp immediate_reply(req) do
-    socket = elem(req, 1)
-    transport = elem(req, 2)
-    {headers, _} = :cowboy_req.headers(req)
-    {_, key} = List.keyfind(headers, "sec-websocket-key", 0)
-
-    challenge =
-      :crypto.hash(:sha, key <> "258EAFA5-E914-47DA-95CA-C5AB0DC85B11") |> Base.encode64()
-
-    handshake =
-      [
-        "HTTP/1.1 101 Test Socket Upgrade",
-        "Connection: Upgrade",
-        "Upgrade: websocket",
-        "Sec-WebSocket-Accept: #{challenge}",
-        "\r\n"
-      ]
-      |> Enum.join("\r\n")
-
-    frame = <<1::1, 0::3, 1::4, 0::1, 15::7, "Immediate Reply">>
-    transport.send(socket, handshake)
-    Process.sleep(0)
-    transport.send(socket, frame)
-
-    Process.sleep(:infinity)
-  end
-end
+  def websocket_info(_, state), do: {:ok, state}
+end
```

Comment:
Please add a new line at the end of the file. 😄

Reply:
Sure!